### PR TITLE
Speed up TF tests by reducing hidden layer counts

### PR DESCRIPTION
**Pull request by Rocketknight1**
A lot of our slow TF tests are caused by TF compilation. TF compilation isn't really affected by layer width at all - the main thing is just the number of operations it has to build a graph for. By reducing the number of hidden layers, compilation gets much faster, (hopefully) without interfering with test coverage at all.

### CODE REVIEW

File: tests/test_modeling_tf_common.py [hunk 1]
```diff
@@ -1527,36 +1527,6 @@ def test_keras_fit(self):
...
-                shuffle=False,
-            )
-            val_loss3 = history3.history["val_loss"][0]
-            self.assertTrue(not isnan(val_loss3))
-            accuracy3 = {key: val[0] for key, val in history3.history.items() if key.endswith("accuracy")}
-            self.check_keras_fit_results(val_loss1, val_loss3)
-            self.assertEqual(history1.history.keys(), history3.history.keys())
-            if metrics:
-                self.assertTrue(len(accuracy1) == len(accuracy3) > 0, "Missing metrics!")
-
```
Comment (Rocketknight1) [inline tests/test_modeling_tf_common.py:1559]: I added this section in `test_keras_fit` when I was paranoid about issues from fitting `tf.data.Dataset`, but I believe it should not be possible for this section to fail if the other model tests pass, so I removed it!

**Comment by ydshieh** [comment]
Would be nice if you can show the timing for one model (before v.s. after) 🙏 . Thanks.

**Comment by Rocketknight1** [comment]
@ydshieh testing locally BERT went from 510 seconds -> 220 seconds

File: tests/models/bert/test_modeling_tf_bert.py [hunk 2]
```diff
@@ -57,7 +57,7 @@ def __init__(
         use_labels=True,
         vocab_size=99,
         hidden_size=32,
-        num_hidden_layers=5,
+        num_hidden_layers=1,
```
Comment (ydshieh) [inline tests/models/bert/test_modeling_tf_bert.py:60]: Let's not use 1 but 2 if possible. `1` is kind exceptional.

Comment (Rocketknight1) [reply tests/models/bert/test_modeling_tf_bert.py]: Fixed!

**Review by ydshieh — APPROVED** [review]
Works for me except not use `1` as num of layers 🙏 .

Let's also have an approval from @sgugger as I am not the one who decide to use `5`.

**Comment by Rocketknight1** [comment]
I don't know if it was @sgugger either - a lot of this code is really old! I see `tf.tuple()` in there, and even I had to look up the TF 1.x docs to remember what that was supposed to do, lol

**Comment by ydshieh** [comment]
I know he is probably not the one to decide use `5`, but he might know the history :-)

**Review by sgugger — APPROVED** [review]
Thanks!

**Pull request merged**